libraries/c_sdk/standard/mqtt/CODESIZE.md

```diff
@@ -0,0 +1,45 @@
+# Code size of MQTT Compatibility Layer
+
+Code size for the MQTT Compatibility Layer is calculated using GCC for ARM for cortex-M4 on [202011.00 release](https://github.com/aws/amazon-freertos/releases/tag/202011.00). Build for calculating code size is done using build configuration for [cypress CY8CKIT-064S0S2-4343W](https://docs.aws.amazon.com/freertos/latest/userguide/getting_started_cypress_psoc64.html) by using the `Release` configuration. All the logs were disabled in this build. Refer to the table below for calculated code sizes.
+
+| File | With -O1 Optimization | With -Os Optimization |
+| :-: | :-: | :-: |
+| iot_mqtt_agent.c | 1.3K | 1.1K |
+| iot_mqtt_api.c | 3.8K | 3.3K |
+| iot_mqtt_context_connection.c | 0.2K | 0.2K |
+| iot_mqtt_managed_function_wrapper.c | 0.9K | 0.9K |
+| iot_mqtt_mutex_wrapper.c | 0.4K | 0.3K |
+| iot_mqtt_network.c | 1.2K | 1.1K |
+| iot_mqtt_operation.c | 2.6K | 2.4K |
+| iot_mqtt_publish_duplicates.c | 0.2K | 0.2K |
+| iot_mqtt_serializer_deserializer_wrapper.c | 2.0K | 1.8K |
+| iot_mqtt_subscription.c | 1.3K | 1.2K |
+| iot_mqtt_subscription_container.c | 0.6K | 0.6K |
+| iot_mqtt_validate.c | 0.5K | 0.4K |
+| **Total estimate** | **15K** | **13.5K** |
+
+The MQTT Compatibility Layer is implemented using [coreMQTT library](https://github.com/FreeRTOS/coreMQTT/blob/master/README.md). In addition to that, this MQTT Compatibility Layer also maintains all the dependencies of MQTT V2.x.x library. In order to account for the total code size cost of this MQTT Compatibility Layer, the code sizes for the dependencies also need to be considered. Please find the code sizes for dependencies below.
```

Comment:
The MQTT Compatibility Layer is implemented using the [coreMQTT library]. In addition, the MQTT Compatibility Layer  maintains all dependencies of MQTT V2.x.x library. To account for the total code size of the MQTT Compatibility Layer, code sizes for all dependencies have been considered (see below).

Reply:
Updated comments. Thanks